env/fs_posix.cc

```diff
@@ -909,7 +906,17 @@ class PosixFileSystem : public FileSystem {
       return IOError("While doing statvfs", fname, errno);
     }
 
-    *free_space = ((uint64_t)sbuf.f_bsize * sbuf.f_bfree);
+    // sbuf.bfree is total free space available to root
+    // sbuf.bavail is total free space available to unprivileged user
+    //  sbuf.bavail <= sbuf.bfree ... pick correct based upon effective user id
+    if (geteuid()) {
+      // non-zero user is unprivileged, or -1 if error.  take more conservative
+      // size
+      *free_space = ((uint64_t)sbuf.f_bsize * sbuf.f_bavail);
```

Comment:
Should this use `f_frsize` instead of `f_bsize`? See cockroachdb/pebble#1072 and https://en.cppreference.com/w/cpp/filesystem/space.

Reply:
@jbowens:  I am using f_bsize because that is what the original Facebook code used in this routine.  I am not going to advocate strongly in either direction since I am not staring at the Linux source code, much less any other operating system's source code.  But my last reference below seems the most strong and supports f_bsize.

I did research and found several things that basically say "seems like both are the same".

This link leans toward f_frsize with some Linux source references:
https://stackoverflow.com/questions/54823541/what-do-f-bsize-and-f-frsize-in-struct-statvfs-stand-for

This link leans toward f_frsize implying fragments that can be less than block size, but also says definitions are vague:
https://unix.stackexchange.com/questions/463369/what-can-f-bsize-be-used-for-is-it-similar-to-st-blksize

Google with search key of "linux f_frsize" list low on the page a link to show a page image from "The Linux Programming Interface: A Linux and UNIX System Programming Handbook".  I have to manual type the paragraph from bottom of page 276: "For most Linux file systems, the values of f_bsize and f_frsize are the same.  However, some file systems support the notion of block fragments, which can be used to allocate a smaller unit of storage at the end of a file if a full block is not required.  This avoids the waste of space that would otherwise occur if a full block was allocated.  On such file systems, f_frsize is the size of a fragment, and f_bsize is the size of a whole block. (The notion of fragments in UNIX file system first appeared in early 1980s with the 4.2BSD Fast File System described in [McKusick et al., 1984].)"

This last quote strongly points me in the direction of f_bsize.  Your mileage may vary.